shell/browser/api/electron_api_net_log.cc

```diff
@@ -178,13 +179,17 @@ void NetLog::StartNetLogAfterCreateFile(net::NetLogCaptureMode capture_mode,
 void NetLog::NetLogStarted(int32_t error) {
   DCHECK(pending_start_promise_);
   ResolvePromiseWithNetError(std::move(*pending_start_promise_), error);
+  pending_start_promise_.reset();
 }
 
 void NetLog::OnConnectionError() {
   net_log_exporter_.reset();
   if (pending_start_promise_) {
+    DCHECK(pending_start_promise_->isolate() !=
+           nullptr);  // std::move(*promise) called without resetting
```

Comment:
It feels arbitrary to have this here and not at lines 164 and 180 above. Is this `DCHECK()` needed? If so, is it needed in each place that `pending_start_promise_` is consumed?